### ISSUE TITLE
Can't use `systemctl edit` with `--global`

### ISSUE THREAD

**Issue by mrc0mmand** [systemd/systemd#31272]
The man page states it should be possible to use `systemctl edit --global` to create a drop-in affecting all users:

https://github.com/systemd/systemd/blob/23f9af646baf743c37dc16133675bf0dffd1cda1/man/systemctl.xml#L1183-L1185

But since d77d42ed3ae this is no longer possible:

```
# echo foo | build/systemctl edit --global --stdin foo.bar
--global is not supported for this operation.
```

/cc @yuwata @YHNdnzj

**Comment by YHNdnzj** [comment]
Hmm, indeed... Currently, `systemctl-edit.c` mixes server- and client-side lookup. Before the commit, when `--global` is specified we actually go through the caller's user manager... I think this would require a more comprehensive fix and just always do client-side instead. Maybe also teach `systemctl edit/cat` to honor `--root`...

**Comment by YHNdnzj** [comment]
(I believe I wanted to do that when I introduced `edit-util` a year ago. But I wasn't too familiar with the codebase and just forgot about it)

**Comment by YHNdnzj** [comment]
Yeah I found this on my todo list - also see https://github.com/systemd/systemd/issues/10974

**Comment by sebastiancarlos** [comment]
Thanks you for tracking this issue! Do you think the  man page could be updated for the time being to match the current behavior?
